ckanext/switzerland/helpers.py

```diff
@@ -470,3 +487,105 @@ def get_cookie_law_url():
+    the server time zone (Europe/Zurich), so they have to be converted to UTC.
+    """
+    for field in CUSTOM_DATETIME_FIELDS:
+        dt = _get_datetime_from_isoformat_string(dataset_or_resource_dict.get(field))
+        if dt is False:
+            continue
+
+        dt_zh = dt.replace(tzinfo=ZURICH)
+        dt_utc = dt_zh.astimezone(UTC)
+        dataset_or_resource_dict[field] = dt_utc.replace(tzinfo=None).isoformat()
+
+    if dataset_or_resource_dict.get("extras"):
+        for extra in dataset_or_resource_dict["extras"]:
+            if extra["key"] in CUSTOM_DATETIME_FIELDS:
+                dt = _get_datetime_from_isoformat_string(extra["value"])
+                if dt is False:
+                    continue
+
+                dt_zh = dt.replace(tzinfo=ZURICH)
+                dt_utc = dt_zh.astimezone(UTC)
+                extra["value"] = dt_utc.replace(tzinfo=None).isoformat()
+
+    if dataset_or_resource_dict.get("resources"):
+        for resource in dataset_or_resource_dict["resources"]:
+            convert_datetimes_for_display(resource)
+
+
+def convert_datetimes_for_api(dataset_or_resource_dict):
+    """Calculates the time of a datetime in the Europe/Zurich time zone and outputs the
+    value as isoformat, with time zone info.
+
+    CKAN stores all datetimes as UTC by default, so they have to be converted to
+    Europe/Zurich and have the time zone info added. For our custom datetime fields, we
+    use the server time zone, so they are already in Europe/Zurich. We just have to add
+    the time zone info.
+    """
+    for field in CKAN_DATETIME_FIELDS:
+        dt = _get_datetime_from_isoformat_string(dataset_or_resource_dict.get(field))
+        if dt is False:
+            continue
+
+        dt_utc = dt.replace(tzinfo=UTC)
+        dt_zh = dt_utc.astimezone(ZURICH)
+        dataset_or_resource_dict[field] = dt_zh.isoformat()
+
+    for field in CUSTOM_DATETIME_FIELDS:
+        dt = _get_datetime_from_isoformat_string(dataset_or_resource_dict.get(field))
+        if dt is False:
+            continue
+
+        dt_zh = dt.replace(tzinfo=ZURICH)
+        dataset_or_resource_dict[field] = dt_zh.isoformat()
+
+    if dataset_or_resource_dict.get("resources"):
+        for resource in dataset_or_resource_dict["resources"]:
+            convert_datetimes_for_api(resource)
+
+
+def request_is_api_request():
+    try:
+        path = tk.request.path
+        if any(
+            [
+                path.endswith(".xml"),
+                path.endswith(".rdf"),
+                path.endswith(".n3"),
+                path.endswith(".ttl"),
+                path.endswith(".jsonld"),
+            ]
+        ):
+            return True
+        if path.startswith("/api") and not path.startswith("/api/action"):
+            # The API client for CKAN's JS modules uses a path starting
+            # /api/action, i.e. without a version number. All other API calls
+            # should include a version number.
```

Comment:
TIL, I assumed that since a few version it is allowed to leave the version out of an standard API call as well, but just tested it with one of our projects and indeed you have to have the version in there, still.